### DISCUSSION TITLE
How to get rich text html from Lexical Text Editor?

### DISCUSSION THREAD

**Discussion by xandemon** [facebook/lexical#4319] [Q&A]
I am currently trying Lexical to implement a rich text editor. With the help of Lexical docs and other tutorials, I am able to create quite a good text editor. Still, the problem is I am just able to make the rich text editor where I can play with all the styles and formatting but I am not being able to get that formatted rich text from the text editor.

For eg: I can get the plain text using the below code:
`console.log(root.getTextContent());`

But how can I get the rich text HTML from Lexical Text Editor? For eg: if there is bold text and italic text in the text editor, I want to get something like the below so that I can use that elsewhere.
Output: `<p><strong>Bold text here</strong></p>`

I searched the documentation and basically the whole internet but I couldn't find any. It could be that I am missing something because I'm comparatively new but I need help.

If possible, I would like you to help me by showing me the code on how I can do that because that would help me better in understanding what you're trying to say.
Thank you!

**Comment by royaheidari2023** [comment]
```
   editorState.read(() => {
        const htmlString = $generateHtmlFromNodes(editor, null);
        console.log("htmlString: ", htmlString);
      });
```

**Reply by ashiq-prospero** [reply]
> editorState

What's wrong with my code, I want to pass [lexicalJson.json](https://github.com/user-attachments/files/20052725/lexicalJson.json) and get html from the hook< i am getting this error `Unable to find an active editor state. State helpers or node methods can only be used synchronously during the callback of editor.update(), editor.read(), or editorState.read()`

```
editor.update(() => {
    const editorState = editor.parseEditorState(JSON.stringify({ ...lexicalJson  }));
    editor.setEditorState(editorState);
});

const editorState = editor.getEditorState();

editorState.read(() => {
    const html = $generateHtmlFromNodes(editor, null);
    console.log(html);
});
```

This dont work either

```
editor.read(() => {
    const html = $generateHtmlFromNodes(editor, null);
    console.log(html);
});
```
